# HG changeset patch
# User Neil <nyamatongwe@gmail.com>
# Date 1489633459 -39600
# Node ID 2b25287311d015ecc9971c101f0f40ae00c8ca4e
# Parent  34665edc7efdd526c62e426bb0f6267f64e156e7
For single phase drawing ensure edges and mark underline drawn over fold tags by
moving drawing of edge line and mark underline after fold tags.

diff -r 34665edc7efd -r 2b25287311d0 src/EditView.cxx
--- a/src/EditView.cxx	Thu Mar 16 13:49:54 2017 +1100
+++ b/src/EditView.cxx	Thu Mar 16 14:04:19 2017 +1100
@@ -1857,8 +1857,10 @@
 
 	if (phase & drawIndicatorsBack) {
 		DrawIndicators(surface, model, vsDraw, ll, line, xStart, rcLine, subLine, lineRange.end, true, model.hoverIndicatorPos);
-		DrawEdgeLine(surface, vsDraw, ll, rcLine, lineRange, xStart);
-		DrawMarkUnderline(surface, model, vsDraw, line, rcLine);
+		if (phasesDraw != phasesOne) {
+			DrawEdgeLine(surface, vsDraw, ll, rcLine, lineRange, xStart);
+			DrawMarkUnderline(surface, model, vsDraw, line, rcLine);
+		}
 	}
 
 	if (phase & drawText) {
@@ -1874,14 +1876,15 @@
 		DrawIndicators(surface, model, vsDraw, ll, line, xStart, rcLine, subLine, lineRange.end, false, model.hoverIndicatorPos);
 	}
 
-	// End of the drawing of the current line
+	DrawFoldDisplayText(surface, model, vsDraw, ll, line, xStart, rcLine, subLine, subLineStart, phase);
+
 	if (phasesDraw == phasesOne) {
 		DrawEOL(surface, model, vsDraw, ll, rcLine, line, lineRange.end,
 			xStart, subLine, subLineStart, background);
+		DrawEdgeLine(surface, vsDraw, ll, rcLine, lineRange, xStart);
+		DrawMarkUnderline(surface, model, vsDraw, line, rcLine);
 	}
 
-	DrawFoldDisplayText(surface, model, vsDraw, ll, line, xStart, rcLine, subLine, subLineStart, phase);
-
 	if (!hideSelection && (phase & drawSelectionTranslucent)) {
 		DrawTranslucentSelection(surface, model, vsDraw, ll, line, rcLine, subLine, lineRange, xStart);
 	}
